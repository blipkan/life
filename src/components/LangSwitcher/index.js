import React, { useMemo } from 'react'
import * as PropTypes from 'prop-types'

import map from 'lodash/map'
import values from 'lodash/values'

import { createCompose } from 'utils/compose'

import { useCurrentLang } from 'services/Language/useCurrentLang'

import './index.scss'

import { Languages } from 'constants/Lang'

export const Defaults = {
  className: 'lang-switcher',
  domId: 'lang-switcher'
}

const LangSwitcher = props => {
  const compose = createCompose(props, Defaults)
  const { langKey, change } = useCurrentLang()

  const items = useMemo(() => {
    return map(values(Languages), descriptor => ({
      descriptor,
      isSelected: langKey === descriptor.key
    }))
  })

  const doItemClick = item => {
    if (item.isSelected) return
    change(item.descriptor.key)
  }

  return (
    <div className={compose.className()} id={compose.id()}>
      {items.map(item => {
        const {
          isSelected,
          descriptor: { key, caption, title }
        } = item

        const className = isSelected
          ? compose.classNameAndModifier('selected', 'item')
          : compose.className('item')

        return (
          <span
            className={className}
            id={compose.id('item')}
            key={key}
            title={title}
            onClick={e => {
              e.stopPropagation()
              doItemClick(item)
            }}
          >
            {caption}
          </span>
        )
      })}
    </div>
  )
}

LangSwitcher.propTypes = {
  domId: PropTypes.string,
  className: PropTypes.string
}

export default LangSwitcher
