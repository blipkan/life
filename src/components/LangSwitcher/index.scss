.lang-switcher {
  position: absolute;
  right: 16px;
  top: 4px;
  display: flex;
  color: #fff;
  max-height: 72px;
  align-items: center;
  flex-flow: row nowrap;
  justify-items: center;
  font-size: 12px;

  &__item {
    justify-items: center;
    align-items: center;
    display: flex;
    vertical-align: center;
    margin-left: 4px;
    background-color: transparent;
    border-radius: 2px;
    padding: 0 4px 2px 4px;
    text-decoration: underline;
    font-weight: 700;

    &_selected {
      background-color: #5a3472;
      text-decoration: none;
      font-weight: normal;
      cursor: auto;
    }
  }
}
