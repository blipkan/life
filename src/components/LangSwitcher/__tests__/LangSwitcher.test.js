/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import LangSwitcher from '../index'

describe('LangSwitcher', () => {
  it('render component', () => {
    const component = shallow(<LangSwitcher />)
    expect(component).toMatchSnapshot()
  })
})
