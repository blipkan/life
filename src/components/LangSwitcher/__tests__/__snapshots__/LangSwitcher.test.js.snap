// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LangSwitcher render component 1`] = `
<div
  className="lang-switcher"
  id="lang-switcher"
>
  <span
    className="lang-switcher__item lang-switcher__item_selected"
    id="lang-switcher_item"
    key="ru"
    onClick={[Function]}
    title="Русский"
  >
    рус
  </span>
  <span
    className="lang-switcher__item"
    id="lang-switcher_item"
    key="en"
    onClick={[Function]}
    title="English"
  >
    en
  </span>
</div>
`;
